app/services/user_alerts/alert_user_about_new_device.rb

```diff
@@ -1,18 +1,36 @@
 module UserAlerts
   class AlertUserAboutNewDevice
     def self.call(user, device, disavowal_token)
-      device_decorator = DeviceDecorator.new(device)
-      login_location = device_decorator.last_sign_in_location_and_ip
-      device_name = device_decorator.nice_name
+      if IdentityConfig.store.feature_new_device_alert_aggregation_enabled
+        user.sign_in_new_device_at ||= Time.zone.now
+        user.save!
+      else
+        device_decorator = DeviceDecorator.new(device)
+        login_location = device_decorator.last_sign_in_location_and_ip
+        device_name = device_decorator.nice_name
 
-      user.confirmed_email_addresses.each do |email_address|
-        UserMailer.with(user: user, email_address: email_address).new_device_sign_in(
-          date: device.last_used_at.in_time_zone('Eastern Time (US & Canada)').
-            strftime('%B %-d, %Y %H:%M Eastern Time'),
-          location: login_location,
-          device_name: device_name,
-          disavowal_token: disavowal_token,
-        ).deliver_now_or_later
+        user.confirmed_email_addresses.each do |email_address|
+          UserMailer.with(user: user, email_address: email_address).new_device_sign_in(
+            date: device.last_used_at.in_time_zone('Eastern Time (US & Canada)').
+              strftime('%B %-d, %Y %H:%M Eastern Time'),
+            location: login_location,
+            device_name: device_name,
+            disavowal_token: disavowal_token,
+          ).deliver_now_or_later
+        end
+      end
+    end
+    if IdentityConfig.store.feature_new_device_alert_aggregation_enabled
+      def self.send_alert(events)
```

Comment:
I wouldn't personally recommend conditionally defining a method like this. The method doesn't do anything anyways so I think it'd be fine to drop the condition altogether. If the method did do something, I'd just put the condition as an early return of the method body.